### ISSUE TITLE
Load Balancer with 2-way SSL authentication support

### ISSUE THREAD

**Issue by grammy-jiang** [getsentry/sentry-python#3256]
### Problem Statement

Hi,

I am configuring Nginx as load balancer in front of Sentry and 2-way SSL authentication is enabled.

I can visit the web GUI as usual after I add the client cert and key to the browser. But the server can't receive the log messages, and I notice that in sentry-python it doesn't support two-way SSL authentication as a client - from the source code no client cert related options are passed to the urllib3.PoolManager:

* [sentry-python/sentry_sdk/transport.py at 69ecd87aa4539de03754af5afb4af4be53efd260 · getsentry/sentry-python](https://github.com/getsentry/sentry-python/blob/69ecd87aa4539de03754af5afb4af4be53efd260/sentry_sdk/transport.py#L542)
* [sentry-python/sentry_sdk/transport.py at 69ecd87aa4539de03754af5afb4af4be53efd260 · getsentry/sentry-python](https://github.com/getsentry/sentry-python/blob/69ecd87aa4539de03754af5afb4af4be53efd260/sentry_sdk/transport.py#L453)

The document of urllib3.PoolManager:

* [Advanced Usage - urllib3 1.26.3 documentation](https://urllib3.readthedocs.io/en/1.26.3/advanced-usage.html#client-certificates)

### Solution Brainstorm

From these lines of options defined, I think it can be configured in the same way:

* [sentry-python/sentry_sdk/transport.py at 69ecd87aa4539de03754af5afb4af4be53efd260 · getsentry/sentry-python](https://github.com/getsentry/sentry-python/blob/69ecd87aa4539de03754af5afb4af4be53efd260/sentry_sdk/transport.py#L477-L482)

   ```python
   options["ca_certs"] = (
       ca_certs  # User-provided bundle from the SDK init
       or os.environ.get("SSL_CERT_FILE")
       or os.environ.get("REQUESTS_CA_BUNDLE")
       or certifi.where()
   )
   ```

   Example:

   ```python
   options["cert_file"] = (
       cert_file  # User-provided bundle from the SDK init
       or os.environ.get("CLIENT_CERT_FILE")
   )
   options["key_file"] = (
       key_file  # User-provided bundle from the SDK init
       or os.environ.get("CLIENT_KEY_FILE")
   )
   ```

**Comment by szokeasaurusrex** [comment]
Hi @grammy-jiang, thanks for the feature suggestion! This is something we can definitely look into adding.

I will put the issue on our internal backlog, but it will likely take us some time to get around to implementing this feature, since we have other higher-priority projects that we are working on at the moment. However, if you would like to contribute a PR, we would be happy to review it and help get it merged.

**Comment by grammy-jiang** [comment]
Thanks for your reply, @szokeasaurusrex .

Yes, this is a quite simple change, I can do it.

**Comment by grammy-jiang** [comment]
I also find another issue about Gitlab Integration with two-way SSL authentication:

Currently, the Gitlab integration in Sentry does not support the Gitlab instance with two-way SSL authentication enabled.

I haven't got a chance to investigate the reason, but I can give it a try.

**Comment by szokeasaurusrex** [comment]
> I also find another issue about Gitlab Integration with two-way SSL authentication

Please raise this issue in the [getsentry/sentry](https://github.com/getsentry/sentry) repo, since it seems like this is related to the Sentry server, not the Sentry Python SDK. The SDK does not integrate with GitLab, only the Sentry server does.

**Comment by szokeasaurusrex** [comment]
Looks like this was fixed in [#3258](https://github.com/getsentry/sentry-python/pull/3258)
